src/main/java/seedu/address/logic/parser/ListCommandParser.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.logic.commands.ListCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.ComparableCategory;
+
+
+/**
+ * Parses input arguments and creates a new ListCommand object
+ */
+public class ListCommandParser implements Parser<ListCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ListCommand
+     * and returns a ListCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ListCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.length() == 0) {
+            return new ListCommand(ComparableCategory.NULL, false);
+        }
+        String[] keywords = trimmedArgs.split("\\s+");
+
+        if (keywords.length == 1) {
+            if (keywords[0].equals("true") || keywords[0].equals("false")) {
+                boolean reverse = keywords[0].equals("true");
+                return new ListCommand(ComparableCategory.NULL, reverse);
+            } else {
+                try {
+                    ComparableCategory category = ComparableCategoryParser.parse(keywords[0]);
+                    return new ListCommand(category, false);
+                } catch (ClassNotFoundException cnf) {
+                    throw new ParseException(
+                            String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListCommand.MESSAGE_USAGE));
+                }
+            }
+        } else if (keywords.length == 2) {
+            try {
+                ComparableCategory category = ComparableCategoryParser.parse(keywords[0]);
+                boolean reverse = keywords[1].equals("true");
```

Comment:
Maybe you can consider adding a prefix for [REVERSE] so that you can use the tokenize method in the ArgumentTokenizer class? I think this will make the command more flexible (i.e. the parameters can come in any order)